cmd/create/machinepool/nodepool.go

```diff
@@ -59,6 +57,21 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		os.Exit(1)
 	}
 
+	// Allow the user to select subnet for a single AZ BYOVPC cluster
+	var subnet string
+	if isBYOVPC(cluster) {
```

Comment:
Do we need this check or all Hypershift clusters are going to be BYOVPC? 
Does the user always provide subnets on Hypershift cluster creation - day 1?

Reply:
All hosted clusters are BYOVPC, so perhaps we can assume this validation.
On Day 1 they can provide the subnets in the similar flow as rosa classic clusters (2 subnets for singleaz, 6 per multiaz) but only private subnets are used.

But I guess on Day 2 user may have new subnets available and they can create new NodePools on those new subnets.